test/scenarios/kusto/output/coredefault/src/azure-cli/azure/cli/command_modules/kusto/generated/custom.py

```diff
@@ -46,28 +46,42 @@ def kusto_cluster_create(client,
                          type_=None,
                          user_assigned_identities=None,
                          no_wait=False):
-    if enable_streaming_ingest is None:
-        enable_streaming_ingest = False
-    if enable_purge is None:
-        enable_purge = False
-    if enable_double_encryption is None:
-        enable_double_encryption = False
     parameters = {}
-    parameters['tags'] = tags
+    if tags is not None:
+        parameters['tags'] = tags
     parameters['location'] = location
     parameters['sku'] = sku
-    parameters['zones'] = zones
-    parameters['trusted_external_tenants'] = trusted_external_tenants
-    parameters['optimized_autoscale'] = optimized_autoscale
-    parameters['enable_disk_encryption'] = enable_disk_encryption
-    parameters['enable_streaming_ingest'] = False if enable_streaming_ingest is None else enable_streaming_ingest
-    parameters['virtual_network_configuration'] = virtual_network_configuration
-    parameters['key_vault_properties'] = key_vault_properties
-    parameters['enable_purge'] = False if enable_purge is None else enable_purge
-    parameters['enable_double_encryption'] = False if enable_double_encryption is None else enable_double_encryption
+    if zones is not None:
+        parameters['zones'] = zones
+    if trusted_external_tenants is not None:
+        parameters['trusted_external_tenants'] = trusted_external_tenants
+    if optimized_autoscale is not None:
+        parameters['optimized_autoscale'] = optimized_autoscale
+    if enable_disk_encryption is not None:
+        parameters['enable_disk_encryption'] = enable_disk_encryption
+    if enable_streaming_ingest is not None:
+        parameters['enable_streaming_ingest'] = enable_streaming_ingest
+    else:
+        parameters['enable_streaming_ingest'] = False
+    if virtual_network_configuration is not None:
+        parameters['virtual_network_configuration'] = virtual_network_configuration
+    if key_vault_properties is not None:
+        parameters['key_vault_properties'] = key_vault_properties
+    if enable_purge is not None:
+        parameters['enable_purge'] = enable_purge
+    else:
+        parameters['enable_purge'] = False
+    if enable_double_encryption is not None:
+        parameters['enable_double_encryption'] = enable_double_encryption
+    else:
+        parameters['enable_double_encryption'] = False
     parameters['identity'] = {}
-    parameters['identity']['type'] = type_
-    parameters['identity']['user_assigned_identities'] = user_assigned_identities
+    if type_ is not None:
+        parameters['identity']['type'] = type_
+    if user_assigned_identities is not None:
+        parameters['identity']['user_assigned_identities'] = user_assigned_identities
+    if len(parameters['identity']) == 0:
+        del parameters['identity']
```

Comment:
how about add some common logic to check all the empty dict in the build result instead of adding so much if XXX is not None. Is it more easier to do ?

Reply:
It is verbose. But I found it's hard to implement an easier way in current logic.